analogdevicesinc/hdl
# docs: Use doctools and add index to library and projects

## PR Description

The extensions have been moved to doc tools.
The source code is available at
https://github.com/analogdevicesinc/doctools
And is installed as before:
(cd docs ; pip install -r requirements.txt --upgrade) Since the package is listed on the requirements.txt file.

How to review this PR:
* Checkout branch
* Run the updated instructions on README.md : 
```
Ensure pip is newer than version 23.
$ pip install pip --upgrade
Install the documentation tools.
$ (cd docs ; pip install -r requirements.txt)
Build the documentation with Sphinx.
$ (cd docs ; make html)
```
* Search generated documentation for regressions.
* Raise concerns, feedback in general

Also, add index to library and projects

## PR Type
- [X] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [X] I have performed a self-review of changes
- [ ] I have compiled all hdl projects and libraries affected by this PR
- [ ] I have tested in hardware affected projects, at least on relevant boards
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe files, Copyright etc)
- [ ] I have not introduced new Warnings/Critical Warnings on compilation
- [ ] I have added new hdl testbenches or updated existing ones
